MoreLinq.Test/UnenumerableList.cs

```diff
@@ -10,12 +10,31 @@ namespace MoreLinq.Test
     /// method of <see cref="IEnumerable"/> - either because they should be using the indexer or because they are
     /// expected to be lazily evaluated.
     /// </summary>
+    public sealed class UnenumerableList<T> : UnenumerableListBase<T>, IList<T>
```

Comment:
Given the revert in c8b4de7113bd4162e1a1e0e862c0445c36a3c4af, this isn't used anymore and so should be reverted to its previous state/implementation too.